scene/gui/texture_button.cpp

```diff
@@ -369,6 +356,28 @@ void TextureButton::set_texture_focused(const Ref<Texture2D> &p_focused) {
 	focused = p_focused;
 };
 
+void TextureButton::_set_texture(Ref<Texture2D> *p_destination, const Ref<Texture2D> &p_texture) {
+	DEV_ASSERT(p_destination);
```

Comment:
It's not  a problem with this PR specifically as this was done in TextureProgressBar too, but DEV_ASSERT isn't very commonly used in the database. Maybe a `EFF_FAIL_COND` would be better for consistency ?

Reply:
Sure thing! Makes sense

Reply:
If you amend this PR, I think you can modify the test in `TextureProgressBar` too. I don't think we need another PR just for that. :)

Reply:
Will do to, will use `ERR_FAIL_NULL`

Reply:
The dev assert was used here, because it's internal method and null can only come from wrong usage. Dev assert checks are only done in dev builds AFAIK, null check in any other build is pointless in this case. The argument is property pointer.